compiler/rustc_errors/src/lib.rs

```diff
@@ -940,14 +940,17 @@ impl HandlerInner {
     }
 
     /// Emit all stashed diagnostics.
-    fn emit_stashed_diagnostics(&mut self) {
+    fn emit_stashed_diagnostics(&mut self) -> ErrorReported {
         let diags = self.stashed_diagnostics.drain(..).map(|x| x.1).collect::<Vec<_>>();
-        diags.iter().for_each(|diag| self.emit_diagnostic(diag));
+        diags.iter().for_each(|diag| {
+            self.emit_diagnostic(diag);
+        });
+        ErrorReported::new_i_know_what_im_doing()
```

Comment:
This should only return `ErrorReported` if `diags` isn't empty.

Also, this (and everything else in this file) doesn't need the unchecked constructor, we're in `rustc_errors`.